runtime/src/serde_snapshot/future.rs

```diff
@@ -12,17 +12,17 @@ type AccountsDbFields = super::AccountsDbFields<SerializableAccountStorageEntry>
 // Serializable version of AccountStorageEntry for snapshot format
 #[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
 pub(super) struct SerializableAccountStorageEntry {
-    id: AppendVecId,
+    id: usize,
     accounts_current_len: usize,
 }
 
 pub trait SerializableStorage {
-    fn id(&self) -> AppendVecId;
+    fn id(&self) -> usize;
```

Comment:
@brooksprumo this is going to cause deserialization issues from the snapshot since this is serialized in the `SerializableAccountStorageEntry` in the `AccountsDbFields` here: https://github.com/solana-labs/solana/blob/3366ac4a5a36c144cf006b1f43d828f1a3c94b1e/runtime/src/serde_snapshot/future.rs#L10 right?

Reply:
I think we should be good as long as we don't have more than 2^32 append vecs. Since the type is staying the same, and during deserialization we remap append vec IDs anyway, I think that'll keep us OK. Does that sound reasonable?

Reply:
The data type we deserialize is not intended to be changing.

Reply:
Having 4B append vecs would definitely destroy this plan.